src/main/java/ua/lviv/iot/novaPoshtaAPI/service/DepartmentService.java

```diff
@@ -0,0 +1,184 @@
+package ua.lviv.iot.novaPoshtaAPI.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import ua.lviv.iot.novaPoshtaAPI.datastorage.dal.DepartmentFileStore;
+import ua.lviv.iot.novaPoshtaAPI.model.Courier;
+import ua.lviv.iot.novaPoshtaAPI.model.Department;
+import ua.lviv.iot.novaPoshtaAPI.model.Parcel;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+
+@Service
+public class DepartmentService {
+
+    @Autowired
+    ParcelService parcelService;
+    @Autowired
+    CourierService courierService;
+    @Autowired
+    DepartmentFileStore departmentFileStore;
+
+    private HashMap<Long, Department> departments = new HashMap<>();
+
+    public List<Department> getAllDepartments() {
+        return new LinkedList<>(this.departments.values());
+    }
+
+    public Department getDepartmentById(Long departmentId) {
+        return this.departments.get(departmentId);
+    }
+
+    public void addDepartment(Department department) {
+        List<Long> parcelIds = new LinkedList<>();
+        department.setParcelIds(parcelIds);
+        this.departments.put(department.getDepartmentId(), department);
+    }
+
+    public void updateDepartment(Department department, Long departmentId) {
+        Department oldDpt = this.departments.get(departmentId);
+        Department newDpt = new Department();
+
+        this.departments.remove(departmentId);
+
+        newDpt.setDepartmentId(oldDpt.getDepartmentId());
+        newDpt.setParcelIds(oldDpt.getParcelIds());
+
+        if (department.getLocation() != null) {
+            newDpt.setLocation(department.getLocation());
+        } else {
+            newDpt.setLocation(oldDpt.getLocation());
+        }
+        if (department.getWorkingHours() != null) {
+            newDpt.setWorkingHours(department.getWorkingHours());
+        } else {
+            newDpt.setWorkingHours(oldDpt.getWorkingHours());
+        }
+
```

Comment:
imagine your application crashed at this point. You dropped the old department while a new one is not stored. hash map has an ability to replace elements with the same key automatically